Detectors/TPC/simulation/src/DigitContainer.cxx

```diff
@@ -13,37 +13,50 @@
 /// \author Andi Mathis, TU München, andreas.mathis@ph.tum.de
 
 #include "TPCSimulation/DigitContainer.h"
-#include "TPCBase/Mapper.h"
 #include <iostream>
 
 using namespace o2::TPC;
 
-void DigitContainer::addDigit(size_t hitID, int cru, int timeBin, int row, int pad, float charge)
+void DigitContainer::addDigit(int eventID, size_t hitID, CRU cru, int timeBin, GlobalPadNumber globalPad, float charge)
 {
-  /// Check whether the container at this spot already contains an entry
-  DigitCRU *result = mCRU[cru].get();
-  if(result != nullptr){
-    mCRU[cru]->setDigit(hitID, timeBin, row, pad, charge);
+  const Sector sector = cru.sector();
+  if(mSectorID == sector) {
+    mSector.setDigit(eventID, hitID, cru, timeBin, globalPad, charge);
   }
-  else{
-    const Mapper& mapper = Mapper::instance();
-    mCRU[cru] = std::make_unique<DigitCRU>(cru, mCommonModeContainer);
-    mCRU[cru]->setDigit(hitID, timeBin, row, pad, charge);
+
+  // stupid fix for now to fill the container
+  if (cru.number() == 11){
+    mOutOfSector.emplace_back(DigitOutOfSector(eventID, hitID, sector, timeBin, globalPad, charge));
   }
-  /// Take care of the common mode
-  mCommonModeContainer.addDigit(cru, timeBin, charge);
 }
 
 
 void DigitContainer::fillOutputContainer(std::vector<o2::TPC::Digit> *output, o2::dataformats::MCTruthContainer<o2::MCCompLabel> &mcTruth,
                                          std::vector<o2::TPC::DigitMCMetaData> *debug, int eventTime, bool isContinuous)
 {
-  for(auto &aCRU : mCRU) {
-    if(aCRU == nullptr) continue;
-    aCRU->fillOutputContainer(output, mcTruth, debug, aCRU->getCRUID(), eventTime, isContinuous);
-    if(!isContinuous) {
-      aCRU->reset();
+  /// \todo Use CRU here instead of the sector!
+  mSector.fillOutputContainer(output, mcTruth, debug, Sector(mSectorID), eventTime, isContinuous);
+}
+
+void DigitContainer::fillOutOfSectorContainer(std::vector<o2::TPC::Digit> *output, o2::dataformats::MCTruthContainer<o2::MCCompLabel> &mcTruth)
+{
+  const static Mapper &mapper = Mapper::instance();
+
+  /// sort according to sector, time bin and global pad
+  std::sort(mOutOfSector.begin(), mOutOfSector.end());
+
+  /// merge digits on the same sector, time bin and global pad
+  for(auto it = mOutOfSector.begin(); it < mOutOfSector.end(); ++it) {
+    const int digiPos = output->size();
+    if(*it == *(it-1)) {
```

Comment:
Will this work for the first digit? Or should {{it}} in the loop start from {{mOutOfSector.begin() + 1}}?

Reply:
For the first digit, the DigitOutOfSector will be directly added to the output container.
I have thoroughly tested this, and could not find any problems.